index.js

```diff
@@ -1,97 +1,226 @@
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log("first time :" + message);
-
-                } else if (notReachable[name] == "true") { // secondtime
-                    notReachable[name] = "notagain"; // we don't want second notification
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log(message);
-                    send(message);
-                }
+                handleDownStatus(name, params.url, responseDate, params.mentionsWhenDown);
             }
         });
-    }, interval);
+    }, params.interval);
+}
+
+function handleSuccessStatus(urlName, urlValue, responseDate) {
+    if (urlContainer[urlName] === undefined) {
+        const message = "First check of this session... Endpoint is running! :beer:";
+        send(message);
+        console.log(`${getFormattedDate(responseDate)} - ${message}`);
+    } else {
+        if (urlErrorContainer[urlName] !== undefined) {
+            restoreLocalErrorAndSend(responseDate, urlName);
+        }
+
+        if (urlContainer[urlName] !== UrlStatus.Up) {
+            const message = `:globe_with_meridians: ${urlName} (${urlValue}) is up and running!`;
+            send(message);
+            console.log(`${getFormattedDate(responseDate)} - ${message}`);
+        }
+    }
+
+    urlContainer[urlName] = UrlStatus.Up;
+}
+
+function handleLocalError(urlName, responseDate) {
+    console.log('Local error, couldn\'t reach endpoint.');
+
+    if (urlErrorContainer[urlName] === undefined) {
+        urlErrorContainer[urlName] = responseDate;
+    }
+}
+
+function handleDownStatus(urlName, urlValue, responseDate, mentionsWhenDown) {
+    const trailingMentions = buildMentionsForDownStatus(mentionsWhenDown);
+    const message = `:warning: Oh no! It looks like ${urlName} (${urlValue}) is down.\n${trailingMentions}`;
+    console.log(`${getFormattedDate(responseDate)} - ${message}`);
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint didn't respond :confused:");
+        urlContainer[urlName] = UrlStatus.Pending;
+        return;
+    }
+
+    if (urlErrorContainer[urlName] !== undefined) {
+        restoreLocalErrorAndSend(responseDate, urlName);
+    }
+
+    if (urlContainer[urlName] === UrlStatus.Up) {
+        // First fail! Try again, maybe this was temporary.
+        urlContainer[urlName] = UrlStatus.Pending;
+    } else if (urlContainer[urlName] === UrlStatus.Pending) {
+        // Two fails in a row! Send message, the endpoint is down.
+        urlContainer[urlName] = UrlStatus.Down;
+        send(message);
+    }
+}
+
+function buildMentionsForDownStatus(mentions) {
+    if (mentions === undefined) return '';
+
+    return mentions.reduce((total, mention, index) => {
+        if (mention.kind === undefined || mention.id === undefined) return total;
```

Comment:
why not replace a pair `kind, id` with a simple `kind: id`?

Reply:
or just drop this at all and inline mentions directly, so people would provide `@userId` or `!subteam^groupId`

Reply:
I think it's a better experience for whoever's using the bot to be able to just pass the ID (which is simple to get), so I'll use `kind: id`